modules/__tests__/Link-test.js

```diff
@@ -557,4 +557,48 @@ describe('A <Link>', () => {
       })
     })
   })
+
+  describe('when the "noHref" prop is specified', function () {
+    class App extends Component {
+      render() {
+        return (
+          <div>
+            <Link to="/hello?the=query#hash" noHref>Link with noHref</Link>
+          </div>
+        )
+      }
+    }
+
+    it('returns an anchor tag without an href', function (done) {
+      render((
+        <Router history={createHistory('/')}>
+          <Route path="/" component={App} />
+        </Router>
+      ), node, function () {
+        const link = node.querySelectorAll('a')[0]
+        expect(link.href).toEqual('')
+        done()
+      })
+    })
+
+    it('should transition correctly on click', done => {
+      const steps = [
+        () => {
+          click(node.querySelector('a'), { button: 0 })
+        },
+        ({ location }) => {
+          expect(location.pathname).toEqual('/hello')
+          expect(location.hash).toEqual('#hash')
+        }
+      ]
+
+      const execNextStep = execSteps(steps, done)
+
+      render((
+        <Router history={createHistory('/hello')} onUpdate={execNextStep}>
+          <Route path="/hello" component={App} />
+        </Router>
+      ), node, execSteps)
```

Comment:
Correction, for the one test above that passes `execSteps` into render, not even the first step will be executed, because `execNextStep` is only added as a callback for `onUpdate`, but nothing updates the module (because `execNextStep` isn't added as a callback to `render`), and so nothing happens at all.